src/test/java/org/opentripplanner/middleware/controllers/api/OtpUserControllerTest.java

```diff
@@ -60,69 +61,70 @@ public static void tearDown() {
-     * and leaves OtpUser.phoneNumber intact.
+     * Check that if for some reason the verification SMS is not sent (invalid format or other error from Twilio),
+     * the request is 400-bad request/500-error, and that the user phone number is unchanged.
      */
-    @Test
-    public void smsRequestShouldSetPendingPhoneNumberOnly() {
-        final String PHONE_NUMBER_TO_VERIFY = "+15555550321";
-        final String PHONE_NUMBER_TO_VERIFY_FORMATTED = "(555) 555-0321";
+    @ParameterizedTest
+    @MethodSource("createBadPhoneNumbers")
+    public void invalidNumbersShouldProduceBadRequest(Map.Entry<String, Integer> testCase) {
+        assumeTrue(testsShouldRun());
+        String badNumber = testCase.getKey();
+        int statusCode = testCase.getValue();
 
         // 1. Request verification SMS.
-        // Note that the result of the request for an SMS does not matter
-        // (e.g. if the SMS service is down, the user's phone number should still be recorded).
-        mockAuthenticatedRequest(
+        // The invalid number should fail the call.
+        HttpResponse<String> response = mockAuthenticatedRequest(
             String.format("api/secure/user/%s/verify_sms/%s",
                 otpUser.id,
-                PHONE_NUMBER_TO_VERIFY
+                badNumber
             ),
             otpUser,
             HttpUtils.REQUEST_METHOD.GET
         );
+        assertEquals(statusCode, response.statusCode());
 
-        // 2. Fetch the newly-created user and check the notificationChannel, phoneNumber and isPhoneNumberVerified fields.
-        // This should be the case regardless of the outcome from above.
-        HttpResponse<String> otpUserWithPhoneResponse = mockAuthenticatedRequest(
+        // 2. Fetch the newly-created user.
+        // The phone number should not be updated.
+        HttpResponse<String> otpUserWithPhoneRequest = mockAuthenticatedRequest(
             String.format("api/secure/user/%s", otpUser.id),
             otpUser,
             HttpUtils.REQUEST_METHOD.GET
         );
-        assertEquals(HttpStatus.OK_200, otpUserWithPhoneResponse.statusCode());
+        assertEquals(HttpStatus.OK_200, otpUserWithPhoneRequest.statusCode());
 
-        OtpUser otpUserWithPhone = JsonUtils.getPOJOFromJSON(otpUserWithPhoneResponse.body(), OtpUser.class);
+        OtpUser otpUserWithPhone = JsonUtils.getPOJOFromJSON(otpUserWithPhoneRequest.body(), OtpUser.class);
         assertEquals(INITIAL_PHONE_NUMBER, otpUserWithPhone.phoneNumber);
-        assertEquals(PHONE_NUMBER_TO_VERIFY, otpUserWithPhone.pendingPhoneNumber);
-        assertEquals(PHONE_NUMBER_TO_VERIFY_FORMATTED, otpUserWithPhone.pendingPhoneNumberFormatted);
+        assertTrue(otpUserWithPhone.isPhoneNumberVerified);
+    }
+
+    private static Set<Map.Entry<String, Integer>> createBadPhoneNumbers() {
+        HashMap<String, Integer> cases = new HashMap<>();
```

Comment:
I kind of like the parameterized pattern that uses `Arguments`. That or the test case approach Evan used in some of the MonitoredTrip tests. 

```java
private static Stream<Arguments> createBadPhoneNumbers() {
        return Stream.of(
            Arguments.of("5555555", HttpStatus.BAD_REQUEST_400),
            Arguments.of("+15555550001", HttpStatus.INTERNAL_SERVER_ERROR_500)
        );
    }
```